.gitignore

```diff
@@ -1,72 +1 @@
-# Compiled class file
```

Comment:
Why remove this stuff? Especially the JetBrains related ignores?

Reply:
FWIW, I usually just use a user `.gitignore` rather than relying on any particular contents of the project

Reply:
Project-specific stuff should go in the repo git ignores, developer (e.g., editor)-specific stuff should go in the developers' global git ignores.  Otherwise we end up duplicating everyone's environment/OS/etc. specific ignores everywhere.

Reply:
*Sigh* maybe you're right @stevenschlansker.  Maybe only truly project-specific stuff belongs in the project.  I.e., even things like Java, or the build system are best kept in the users's.  But if you had, e.g., some artifact file that was specifically generated just for your project, then that'd go in the project ignore.

Reply:
I think it would be great if we had a common user-level .gitignore. Unfortunately I'm not sure such a thing exists, so newcomers to a project will just have to be extra diligent that they don't accidentally check in IDE files. I don't see anything particularly wrong with a common per-project .gitignore (which is a perfectly common thing to do).

In any case, this change was irrelevant and out of scope to this changeset. The ignores were already there, so there's little harm in keeping them around.

Reply:
For the record, I use [github/gitignore](https://github.com/github/gitignore) as my source for gitignore boilerplate.

Reply:
@SegFaultAX It's not irrelevant or out of scope; I was generally cleaning up the project as well (e.g., adding readme, changelog).  The harm is in having irrelevant content, which is a distraction when reading and editing the file.  Just because something's common doesn't mean it's good.

I like your idea about a helpful common user-level git ignore.  Maybe that's something we could try to promote here!